File: 4-web-apps/javascript-ajax-apis/foursquare/foursquare-explore.py
import json
import requests

url = 'https://api.foursquare.com/v2/venues/search'

params = dict(client_id='PASTE_YOUR_KEY_HERE',
              client_secret='PASTE_YOUR_SECRET_HERE',
              v='20180323',
              ll='42.3495694,-71.0836727',
              query='george howell',
              limit=1)
resp = requests.get(url=url, params=params)
data = json.loads(resp.text)

# If this file is called as a standalone program:
if __name__ == '__main__':
    # Run a sample location for debugging
    print(data)
    print('\n Venue name: {} \n Venue address: {}'
          .format(data['response']['venues'][0]['name'],
                  data['response']['venues'][0]['location']['address']))

File: 4-web-apps/javascript-ajax-apis/foursquare/foursquare-explore.js
const request = require('request')

request({
  url: 'https://api.foursquare.com/v2/venues/explore',
  method: 'GET',
  qs: {
    client_id: 'PASTE_YOUR_KEY_HERE',
    client_secret: 'PASTE_YOUR_SECRET_HERE',
    ll: '42.3495694,-71.0836727',
    query: 'george howell',
    v: '20180323',
    limit: 1
  }
}, function (err, res, body) {
  if (err) {
    console.error(err)
  } else {
    console.log(body)
  }
})
